### DISCUSSION TITLE
understanding pruning and disk usage

### DISCUSSION THREAD

**Discussion by rutledj** [nextgenhealthcare/connect#5225] [Q&A]
I'm looking for a better understanding of the pruning in Mirth that runs on an appliance. 

In the channel you have Prune metadata older than x days. Understand that. I have that set for every channel.

But at what point should I see my disk size start increasing again. Pruning messages alone doesn't seem to help reclaim disk space as seen on the appliance home page.

**Comment by pacmano1** [comment]
Since the underlying DB is postgres you should see some related answers on the forums     or here.  Or just Google "Why does postgres not reclaim disk space when rows are deleted?".

**Comment by rjkroll** [comment]
Pruning does DELETE statements which inherently doesn't reclaim disk space. But if you regularly prune then it shouldn't matter, and in general your database shouldn't be perpetually growing, unless your message throughput is also growing.

Some databases like PostgreSQL have auto-vacuuming features that you can use, while something like Derby you would have to compress the tables.

After you prune the messages and then vacuum the tables, space should be freed up for internal reuse, but will probably not be handed back to the OS (for use in different files, or to show up in df).

 You can use [pg_freespacemap](https://www.postgresql.org/docs/current/pgfreespacemap.html) or maybe [pgstattuple](https://www.postgresql.org/docs/current/pgstattuple.html) to check if space is available for internal reuse, or pg_stat_activity or [pg_stat_progress_vacuum](https://www.postgresql.org/docs/current/progress-reporting.html) to see if vacuuming is currently underway. 

Vacuuming should happen automatically after a large fraction of a table is deleted, unless you have gone out of your way to prevent it from happening.

**Comment by rutledj** [comment]
So does the disk gauge on the appliance indicate the amount of space being used by the database or mirth connect (or both)?

**Comment by jonbartels** [comment]
https://gist.github.com/MichaelLeeHobbs/67980d165fc68880eb2ab283c673244b is a great script to help manage DB size for your channels It shows the channel information and what categories of data are using the most DB space.

Zen has a video at https://www.youtube.com/watch?v=oc-tLvQ2qkA

For Postgres - set your pruner batch size to 99. Details at https://github.com/nextgenhealthcare/connect/discussions/4691#discussioncomment-1250261 

For more specific help for your circumstances you should share:
* The output from the query I posted above, obscuring channel names if they expose customer or client information
* A screenshot from Settings of your pruner settings for the pruner job (not individual channels)

Finally - You have an MC appliance, you pay licensing fees for that. Even if you have a limited number of support cases, you should have unlimited access to the  NextGen support community and the knowledge base there. Get an account and look up the NG customer information. You might also consider the "health check" services offered by Zen, Innovar, or NextGen.
